File: src/CSPAF/CSPAgents/CSPUtilityServerGroups/CSPCreateUtilityServerGroup.cs
using System;
using System.Collections.Generic;

using CloudSuitePortal;
using CloudSuitePortal.Models;
using CloudSuitePortal.Services;
using CloudSuitePortal.Controllers;

namespace actions.CSP.CSPAgents.CSPUtilityServerGroups
{
    class CreateUtilityServerGroup
    {
        public void run(Dictionary<string, object> Params)
        {
            var name = (string)Params["Name"];
            var region = (string)Params["Region"];
            var description = (string)Params["Description"];
            var amazonCredentials = (string)Params["Amazon Credentials"];
            
            var utilityServerGroup = new CSPUtilityServerGroup();
            utilityServerGroup.Name = name;
            utilityServerGroup.Region = region;
            utilityServerGroup.Description = description;
            utilityServerGroup.AmazonCredentials = amazonCredentials;
            
            var navigationController = CSPBrowser.Application.WindowController.NavigationController;
            var controller = (CSPUtilityServerGroupsController)navigationController.VisibleViewController;
            controller.Create(utilityServerGroup);
        }
    }
}

File: src/CSPAF/CSPAgents/CSPUtilityServerGroups/CSPCreateUtilityServerGroup.java
package CSPAF.CSPAgents.CSPUtilityServerGroups;

import java.util.*;

import com.infor.cloudsuite.portal.core.services.*;
import com.infor.cloudsuite.portal.core.controllers.*;
import com.infor.cloudsuite.portal.singletenant.agents.models.*;
import com.infor.cloudsuite.portal.singletenant.agents.controllers.*;

class CSPCreateUtilityServerGroup
{
    public void run(HashMap<String, Object> params)
    {
        String name = params.get("Name").toString();
        String region = params.get("Region").toString();
        String description = params.get("Description").toString();
        String amazonCredentials = params.get("Amazon Credentials").toString();
            
        CSPUtilityServerGroup utilityServerGroup = new CSPUtilityServerGroup();
        utilityServerGroup.name = name;
        utilityServerGroup.region = region;
        utilityServerGroup.description = description;
        utilityServerGroup.amazonCredentials = amazonCredentials;
            
        CSPNavigationController navigationController = CSPBrowser.getApplication().getWindowController().getNavigationController();
        CSPUtilityServerGroupsController controller = (CSPUtilityServerGroupsController)navigationController.getVisibleViewController();
        controller.create(utilityServerGroup);
    }
}